File: scnserver/models/subscription.go
package models

// [!] subscriptions are read-access to channels,
//
// The set of subscriptions specifies which messages the ListMessages() API call returns
// also single messages/channels that are subscribed can be queries
//
// (use keytokens for write-access)

type Subscription struct {
	SubscriptionID      SubscriptionID `db:"subscription_id"          json:"subscription_id"`
	SubscriberUserID    UserID         `db:"subscriber_user_id"       json:"subscriber_user_id"`
	ChannelOwnerUserID  UserID         `db:"channel_owner_user_id"    json:"channel_owner_user_id"`
	ChannelID           ChannelID      `db:"channel_id"               json:"channel_id"`
	ChannelInternalName string         `db:"channel_internal_name"    json:"channel_internal_name"`
	TimestampCreated    SCNTime        `db:"timestamp_created"        json:"timestamp_created"`
	Confirmed           bool           `db:"confirmed"                json:"confirmed"`
}

File: flutter/lib/models/subscription.dart
class Subscription {
  final String subscriptionID;
  final String subscriberUserID;
  final String channelOwnerUserID;
  final String channelID;
  final String channelInternalName;
  final String timestampCreated;
  final bool confirmed;

  const Subscription({
    required this.subscriptionID,
    required this.subscriberUserID,
    required this.channelOwnerUserID,
    required this.channelID,
    required this.channelInternalName,
    required this.timestampCreated,
    required this.confirmed,
  });

  factory Subscription.fromJson(Map<String, dynamic> json) {
    return Subscription(
      subscriptionID: json['subscription_id'] as String,
      subscriberUserID: json['subscriber_user_id'] as String,
      channelOwnerUserID: json['channel_owner_user_id'] as String,
      channelID: json['channel_id'] as String,
      channelInternalName: json['channel_internal_name'] as String,
      timestampCreated: json['timestamp_created'] as String,
      confirmed: json['confirmed'] as bool,
    );
  }

  static List<Subscription> fromJsonArray(List<dynamic> jsonArr) {
    return jsonArr.map<Subscription>((e) => Subscription.fromJson(e as Map<String, dynamic>)).toList();
  }
}
